rust/vectordb/src/data/inspect.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2023 Lance Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use arrow::compute::kernels::{aggregate::bool_and, length::length};
+use arrow_array::{cast::AsArray, types::Int32Type, RecordBatchReader};
+use arrow_ord::comparison::eq_dyn_scalar;
+use arrow_schema::DataType;
+
+use crate::error::{Error, Result};
+
+/// Infer the vector columns from a dataset.
+///
+/// Parameters
+/// ----------
+/// - reader: RecordBatchReader
+/// - strict: if set true, only fixed_size_list<float> is considered as vector column. If set to false,
```

Comment:
I'm wondering if we should have a minimum length requirement here. I guess UMAP/PCA results can be low dimensions, but I also worry about false positives?

Reply:
32? 64? I dont have a good sense where to cut the line now.